components/eam/src/dynamics/se/se_single_column_mod.F90

```diff
@@ -505,4 +510,150 @@ subroutine iop_domain_relaxation(elem,hvcoord,hybrid,t1,dp,nelemd_todo,np_todo,d
 
+subroutine crm_resolved_turb(elem,hvcoord,hybrid,t1,&
+                              nelemd_todo,np_todo)
+
+  ! Subroutine intended to compute various resolved turbulence statistics
+  ! (done so to be consistent with SHOC's definition of each for ease of
+  ! comparison) for DP CRM runs.
+
+  use scamMod
+  use dimensions_mod, only : np, np, nlev, nlevp, npsq, nelem
+  use parallel_mod, only: global_shared_buf, global_shared_sum
+  use global_norms_mod, only: wrap_repro_sum
+  use hybvcoord_mod, only : hvcoord_t
+  use hybrid_mod, only : hybrid_t
+  use element_mod, only : element_t
+  use time_mod
+  use physical_constants, only : Cp, Rgas
+  use cam_history, only: outfld
+  use physconst, only: latvap
+
+  ! Input/Output variables
+  type (element_t)     , intent(inout), target :: elem(:)
+  type (hvcoord_t)                  :: hvcoord
+  type(hybrid_t),             intent(in) :: hybrid
+  integer, intent(in) :: nelemd_todo, np_todo, t1
+
+  ! Local variables
+  real (kind=real_kind), dimension(nlev) :: domain_thetal, domain_qw, domain_u, domain_v
+  real (kind=real_kind) :: thetal(np,np,nlev), qw(np,np,nlev)
+  real (kind=real_kind) :: rho(np,np,nlev), temperature(np,np,nlev)
+  real (kind=real_kind) :: wthl_res(np,np,nlev+1), wqw_res(np,np,nlev+1)
+  real (kind=real_kind) :: w2_res(np,np,nlev+1), w3_res(np,np,nlev+1)
+  real (kind=real_kind) :: u2_res(np,np,nlev), v2_res(np,np,nlev)
+  real (kind=real_kind) :: thl2_res(np,np,nlev), qw2_res(np,np,nlev), qwthl_res(np,np,nlev)
+  real (kind=real_kind) :: pres(nlev)
+  integer :: ie, i, j, k
+
+  ! Compute pressure for IOP observations
+  do k=1,nlev
+    pres(k) = hvcoord%hyam(k)*hvcoord%ps0 + hvcoord%hybm(k)*psobs
+  end do
+
+  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  ! Find the domain mean of thetal, qw, u, and v
+  do k=1,nlev
+    do ie=1,nelemd_todo
+
+      ! Compute potential temperature
+      call get_pottemp(elem(ie),thetal,hvcoord,t1,1)
+
+      ! Compute liquid water potential temperature
+      thetal(:,:,k) = thetal(:,:,k) - (latvap/Cp)*(elem(ie)%state%Q(1:np,1:np,k,2))
+      ! Compute total water
+      qw(:,:,k) = elem(ie)%state%Q(1:np,1:np,k,1) + elem(ie)%state%Q(1:np,1:np,k,2)
+
+      ! Initialize the global buffer
+      global_shared_buf(ie,1) = 0.0_real_kind
+      global_shared_buf(ie,2) = 0.0_real_kind
+      global_shared_buf(ie,3) = 0.0_real_kind
+      global_shared_buf(ie,4) = 0.0_real_kind
+
+      ! Sum each variable for each level
+      global_shared_buf(ie,1) = global_shared_buf(ie,1) + SUM(thetal(:,:,k))
+      global_shared_buf(ie,2) = global_shared_buf(ie,2) + SUM(qw(:,:,k))
+      global_shared_buf(ie,3) = global_shared_buf(ie,3) + SUM(elem(ie)%state%v(:,:,1,k,t1))
+      global_shared_buf(ie,4) = global_shared_buf(ie,4) + SUM(elem(ie)%state%v(:,:,2,k,t1))
+    enddo
+
+    ! Compute the domain mean
+    call wrap_repro_sum(nvars=4, comm=hybrid%par%comm)
+    domain_thetal(k) = global_shared_sum(1)/(dble(nelem)*dble(np)*dble(np))
```

Comment:
It looks like each column is weighted equally. my impression was that some nodes were larger than others. If so, do the variables need to be area-weighted in this case?

Reply:
For the planar grid (which these diagnostics are specific for, for DP-SCREAM), this is not an issue.